optimum/onnxruntime/model.py

```diff
@@ -62,6 +65,7 @@ def __init__(
         self.onnx_named_inputs = list(onnx_config.inputs.keys())
         self.onnx_named_outputs = list(onnx_config.outputs.keys())
         self.onnx_config = onnx_config
+        self.execution_provider = execution_provider
```

Comment:
An other thing that could be added is to verify that the given `execution_provider` is in the available providers list `onnxruntime.get_available_providers()`

Reply:
I think we can we can safely omit this check, as onnxruntime already throws a nice error message when `InferenceSession()` is called with an unavailable execution provider:

```
.../site-packages/onnxruntime/capi/onnxruntime_inference_collection.py:55: UserWarning: Specified provider 'CUDAExecutionProvider' is not in available provider names.Available providers: 'CPUExecutionProvider'
```

What do you think?